File: sdks/python/examples/basic_usage.py
"""
Basic usage example for Kosha Python Client SDK.

Demonstrates:
- Client initialization
- Health check
- Single transaction processing
- Basic error handling
"""

import sys
from datetime import UTC, datetime

# Add parent directory to path for local imports
sys.path.insert(0, "..")

from kosha_client import KoshaClient


def main():
    """Demonstrate basic SDK usage."""
    print("=" * 60)
    print("Kosha Client SDK - Basic Usage Example")
    print("=" * 60)

    # Initialize client
    client = KoshaClient(api_url="http://127.0.0.1:8000", timeout=30)

    # 1. Health Check
    print("\n1. Checking API health...")
    try:
        health = client.health_check()
        print(f"   ✓ API Status: {health['status']}")
        print(f"   ✓ Model Loaded: {health.get('model_loaded', 'unknown')}")
    except Exception as e:
        print(f"   ✗ Health check failed: {e}")
        print("\n   Make sure the API is running:")
        print("   uvicorn app.main:app --reload")
        sys.exit(1)

    # 2. Process a single transaction
    print("\n2. Processing single transaction...")
    transaction = {
        "client_id": "DEMO_CLIENT",
        "transaction_id": "DEMO_TXN_001",
        "sequence_id": 1,
        "event_timestamp": datetime.now(UTC).isoformat(),
        "execution_timestamp": datetime.now(UTC).isoformat(),
        "ledger_a_amount": 1000.0,
        "ledger_a_fx": 1.2345,
        "ledger_a_fee": 10.0,
        "ledger_b_amount": 1000.0,
        "ledger_b_fx": 1.2345,
        "ledger_b_fee": 10.0,
        "currency": "USD",
        "vendor_name": "Acme Corporation",
        "counterparty_name": "Acme Corporation",
        "source_system_id": "DEMO_SYSTEM",
        "context_data": {"demo": True},
    }

    try:
        result = client.reconcile_transaction(transaction)
        print("   ✓ Transaction processed successfully")
        print(f"   ✓ Transaction ID: {transaction['transaction_id']}")
        print(f"   ✓ Exception Flag: {result['exception_flag']}")
        print(f"   ✓ Reason Code: {result['reason_code']}")
        print(f"   ✓ Confidence: {result['confidence']:.2%}")
        print(f"   ✓ Audit Hash: {result['audit_hash'][:16]}...")
    except Exception as e:
        print(f"   ✗ Transaction processing failed: {e}")
        sys.exit(1)

    print("\n" + "=" * 60)
    print("✓ Basic usage example completed successfully")
    print("=" * 60)


if __name__ == "__main__":
    main()

File: examples/getting-started/basic-usage.ts
/**
 * Basic usage example for Kosha Finance TypeScript Client SDK.
 *
 * Demonstrates:
 * - Client initialization
 * - Health check
 * - Single transaction processing
 * - Basic error handling
 */

import { KoshaClient } from '@kosha-finance/client';

async function main() {
  console.log('='.repeat(60));
  console.log('Kosha Finance Client SDK - Basic Usage Example');
  console.log('='.repeat(60));

  // Initialize client
  const client = new KoshaClient({
    apiUrl: 'https://api.kosha.finance',
    apiKey: process.env.KOSHA_API_KEY || 'your_api_key_here',
    timeout: 30000
  });

  // 1. Health Check
  console.log('\n1. Checking API health...');
  try {
    const health = await client.healthCheck();
    console.log(`   ✓ API Status: ${health.status}`);
    console.log(`   ✓ Model Loaded: ${health.model_loaded ?? 'unknown'}`);
  } catch (error) {
    console.error(`   ✗ Health check failed: ${error}`);
    console.error('\n   Make sure the API is accessible and your API key is valid');
    process.exit(1);
  }

  // 2. Process a single transaction
  console.log('\n2. Processing single transaction...');
  const transaction = {
    amount: 100.50,
    currency: 'USD',
    transaction_date: new Date().toISOString().split('T')[0],
    description: 'Payment to Acme Corporation',
    merchant_name: 'Acme Corporation',
    transaction_type: 'debit',
    category: 'business_expense'
  };

  try {
    const result = await client.reconcileTransaction(transaction);
    console.log('   ✓ Transaction processed successfully');
    console.log(`   ✓ Match Found: ${result.is_match}`);
    console.log(`   ✓ Match Type: ${result.match_type}`);
    console.log(`   ✓ Confidence: ${(result.confidence * 100).toFixed(2)}%`);
    if (result.match_details) {
      console.log(`   ✓ Match Details:`, JSON.stringify(result.match_details, null, 2));
    }
  } catch (error) {
    console.error(`   ✗ Transaction processing failed: ${error}`);
    process.exit(1);
  }

  console.log('\n' + '='.repeat(60));
  console.log('✓ Basic usage example completed successfully');
  console.log('='.repeat(60));
}

// Run the example
main().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});
